src/common/json_encode.c

```diff
@@ -88,3 +88,39 @@ const char *jsonify_bool(bool value) {
         return json_false;
     }
 }
+
+void replace_first_char(char *data, size_t data_size, char replace_with, size_t to_replace_size) {
+    data[0] = replace_with;
+    // shift the rest of the string by the amount of replaced chars
+    memmove(data + 1, data + to_replace_size, data_size - (to_replace_size));
```

Comment:
This'll make the whole de-escaping algorithm an `O(n²)`  ☹. Can't you make it that you have a „read pointer“ and „write pointer“ (or offset or whatever) and shovel the decoded / read ones to the write one?